test/bricks-net-api.test.js

```diff
@@ -0,0 +1,634 @@
+						}
+						catch (ex) {
+							done(ex);
+						}
+					});
+				}
+				catch (ex) {
+					done(ex);
+				}
+			};
+			
+			var response = api.HTTP(
+				api.POST("localhost:20000/test", requestObject)
+			);
+			
+			when(
+				response.body
+			).then(function (body) {
+				assert.equal('string', typeof body, '`body` is a string');
+				assert.equal(requestJson, body, '`body` equals to the response body');
+				done();
+			}).done(undefined, done);
+		});
+		
+		it('should disallow redirects by default', function (done) {
+			serverRequestHandler = function (request, response) {
+				serverRequestHandler = function (request, response) {
+					response.end('OK');
+				};
+				
+				response.writeHead(302, { "Location": '/to' });
+				response.end();
+			};
+			
+			var response = api.HTTP(api.GET("localhost:20000/from"));
+			
+			assert.equal(true, response instanceof api.HTTPResponse, 'response instanceof HTTPResponse');
+			assert.equal(true, when.isPromiseLike(response), 'isPromiseLike');
+			
+			when(
+				response.body
+			).then(function (body) {
+				done(new Error('Resolved instead of rejected.'));
+			}, function (err) {
+				assert.equal(true, err instanceof HTTPRedirectNotAllowedException);
+				done();
+			}).done(undefined, done);
+		});
+		
+		it('should handle redirects if allowed', function (done) {
+			serverRequestHandler = function (request, response) {
+				serverRequestHandler = function (request, response) {
+					response.end('OK');
+				};
+				
+				response.writeHead(302, { "Location": '/to' });
+				response.end();
+			};
+			
+			var response = api.HTTP(api.GET("localhost:20000/from").AllowRedirects());
+			
+			assert.equal(true, response instanceof api.HTTPResponse, 'response instanceof HTTPResponse');
+			assert.equal(true, when.isPromiseLike(response), 'isPromiseLike');
+			
+			when(
+				response.body
+			).then(function (body) {
+				assert.equal('string', typeof body, '`body` is a string');
+				assert.equal('OK', body, '`body` equals to the response body');
+				done();
+			}).done(undefined, done);
+		});
+		
+		it('should handle redirect loop', function (done) {
+			var redirectHandler1 = function (request, response) {
+				serverRequestHandler = redirectHandler2;
+				
+				response.writeHead(302, { "Location": '/redirect2' });
+				response.end();
+			};
+			var redirectHandler2 = function (request, response) {
+				serverRequestHandler = redirectHandler3;
+				
+				response.writeHead(302, { "Location": '/redirect3' });
+				response.end();
+			};
+			var redirectHandler3 = function (request, response) {
+				serverRequestHandler = redirectHandler1;
+				
+				response.writeHead(302, { "Location": '/redirect1' });
+				response.end();
+			};
+			
+			serverRequestHandler = redirectHandler1;
+			
+			var response = api.HTTP(api.GET("localhost:20000/redirect1").AllowRedirects());
+			
+			assert.equal(true, response instanceof api.HTTPResponse, 'response instanceof HTTPResponse');
```

Comment:
Do you need the 3rd parameter for these `assert`-s? The line number would be rather obvious, won't it?